src/marqo/core/search/hybrid_search.py

```diff
@@ -0,0 +1,212 @@
+            raise core_exceptions.UnsupportedFeatureError(
+                "Unstructured indexes are not yet supported for hybrid search. "
+                "Please use a structured index.")
+
+        marqo_index_version = marqo_index.parsed_marqo_version()
+        if marqo_index_version < constants.MARQO_HYBRID_SEARCH_MINIMUM_VERSION:
+            raise core_exceptions.UnsupportedFeatureError(
+                f"Hybrid search is only supported for Marqo indexes with version >= {constants.MARQO_HYBRID_SEARCH_MINIMUM_VERSION}. "
+                f"Current index version is {marqo_index_version}."
+            )
+
+        # Use default hybrid settings if not provided
+        if hybrid_parameters is None:
+            hybrid_parameters = HybridParameters()
+
+        # Determine the text query prefix
+        text_query_prefix = marqo_index.model.get_text_query_prefix(text_query_prefix)
+
+        if isinstance(query, CustomVectorQuery):
+            query_text_vectorise = None
+            query_text_search = query.custom_vector.content
+
+            if context is None:
+                context = SearchContext(
+                    tensor=[SearchContextTensor(vector=query.custom_vector.vector, weight=1)]
+                )
+            else:
+                context.tensor.append(SearchContextTensor(vector=query.custom_vector.vector, weight=1))
+        else:  # string query
+            query_text_vectorise = query
+            query_text_search = query
+
+        queries = [BulkSearchQueryEntity(
+            q=query_text_vectorise, searchableAttributes=searchable_attributes, searchMethod=SearchMethod.HYBRID,
+            limit=result_count,
+            offset=offset, showHighlights=False, filter=filter_string, attributesToRetrieve=attributes_to_retrieve,
+            boost=boost, image_download_headers=image_download_headers, context=context, scoreModifiers=score_modifiers,
+            index=marqo_index, modelAuth=model_auth, text_query_prefix=text_query_prefix,
+            hybridParameters=hybrid_parameters
+        )]
+
+        with RequestMetricsStore.for_request().time(f"search.hybrid.vector_inference_full_pipeline"):
+            qidx_to_vectors: Dict[Qidx, List[float]] = run_vectorise_pipeline(config, queries, device)
+        vectorised_text = list(qidx_to_vectors.values())[0]
+
+        # Parse text into required and optional terms.
+        (required_terms, optional_terms) = utils.parse_lexical_query(query_text_search)
+
+        marqo_query = MarqoHybridQuery(
+            index_name=index_name,
+            vector_query=vectorised_text,
+            filter=filter_string,
+            limit=result_count,
+            ef_search=ef_search,
+            approximate=approximate,
+            offset=offset,
+            or_phrases=optional_terms,
+            and_phrases=required_terms,
+            attributes_to_retrieve=attributes_to_retrieve,
+            searchable_attributes=searchable_attributes,
```

Comment:
In the definition of `MarqoHybridQuery` we validate both `searchable_attributes` and `score_modifiers` should not be set. So why do we set them here? just to trigger the validation error?

Reply:
Yes, just to trigger the validation error